src/main/java/net/masterthought/cucumber/json/deserializers/StepDeserializer.java

```diff
@@ -0,0 +1,65 @@
+package net.masterthought.cucumber.json.deserializers;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import net.masterthought.cucumber.json.DocString;
+import net.masterthought.cucumber.json.Embedding;
+import net.masterthought.cucumber.json.Match;
+import net.masterthought.cucumber.json.Output;
+import net.masterthought.cucumber.json.Result;
+import net.masterthought.cucumber.json.Row;
+import net.masterthought.cucumber.json.Step;
+
+public class StepDeserializer extends JsonDeserializer<Step> {
+
+    @Override
+    public Step deserialize(JsonParser parser, DeserializationContext context) throws IOException {
+        ObjectMapper mapper = (ObjectMapper) parser.getCodec();
+        JsonNode rootNode = mapper.readTree(parser);
+
+        String name = rootNode.get("name").asText();
+        String keyword = rootNode.get("keyword").asText();
+        Result result = mapper.convertValue(rootNode.get("result"), Result.class);
+        Row[] rows = mapper.convertValue(rootNode.get("rows"), Row[].class) == null ?
+                new Row[0] : mapper.convertValue(rootNode.get("rows"), Row[].class);
+        Match match = mapper.convertValue(rootNode.get("match"), Match.class);
+        Embedding[] embeddings = mapper.convertValue(rootNode.get("embeddings"), Embedding[].class);
+        DocString docString = mapper.convertValue(rootNode.get("doc_string"), DocString.class);
+        Output[] outputs = parseOutputs(rootNode.get("output"));
+
+        return new Step(name, keyword, result, rows, match, embeddings, docString, outputs);
+    }
```

Comment:
No longer relevant. This code was deleted after creating deserializer only for output node.